src/script/event/EventRepository.ts

```diff
@@ -640,12 +642,12 @@ export class EventRepository {
   private getUpdatesForMessage(originalEvent: EventRecord, newEvent: MessageAddEvent) {
     const newData = newEvent.data;
     const originalData = originalEvent.data;
-    const updatingLinkPreview = !!originalData.previews.length;
+    const updatingLinkPreview = !!(originalData as any).previews.length;
     if (updatingLinkPreview) {
       this.throwValidationError(newEvent, 'ID of link preview reused');
     }
 
-    const textContentMatches = !newData.previews?.length || newData.content === originalData.content;
+    const textContentMatches = !newData.previews?.length || newData.content === (originalData as any).content;
```

Comment:
I have a followup PR that will extract all this code to a new module. This will be fixed there